### DISCUSSION TITLE
gnutls: Xcode 12 seems to break building Guile extensions

### DISCUSSION THREAD

**Discussion by sharris40** [Homebrew/discussions#38] [Everyday usage]
I believe the postinstall script for gnutls will fail when building from source on Xcode 12 due to a bug in the configure script. The script checks if the Guile function `scm_from_locale_string` is available in the libraries, but the check doesn't include the prototype from libguile/strings.h. I'm not sure if this is intentional due to past changes in Guile's header structures, but in any case, the check now fails due to implicit function declarations now being an error. This causes the Guile extensions to be disabled and the postinstall script to fail.

I fixed it for myself by adding the following patch and running autoreconf before configure. As I said, I do not know if this would be appropriate for upstream, or even for Homebrew, so I have not filed an issue.

    --- configure.ac	2020-09-04 01:37:27.000000000 -0500
    +++ configure.ac	2020-10-07 16:47:03.000000000 -0500
    @@ -1094,7 +1094,7 @@
           CFLAGS="$CFLAGS $GUILE_CFLAGS"
           LIBS="$LIBS $GUILE_LDFLAGS"
           AC_MSG_CHECKING([whether GNU Guile is recent enough])
    -      AC_LINK_IFELSE([AC_LANG_PROGRAM([], [scm_from_locale_string ("")])],
    +      AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <libguile/strings.h>], [scm_from_locale_string ("")])],
             [], [opt_guile_bindings=no])
           CFLAGS="$save_CFLAGS"
           LIBS="$save_LIBS"

**Comment by SMillerDev** [comment]
All bugs should be reported upstream first.
